docs/src/integration_test_tool.md

```diff
@@ -1,3 +1,161 @@
 # Integration Test Tool
 
-Coming soon.
+The public API `IntegrationTests.jl` only provides a function called `depending_projects()`, which returns a vector of package names that use the package you are looking for (`package_name`). In addition to the package name searched for, the function also requires a package name filter (`package_filter`) and a project dependency tree (`project_tree`). The project dependency can be constructed from the currently active Julia project environment, which is highly recommended.
+
+```@docs
+depending_projects(package_name::AbstractString,
+    package_filter::Union{<:AbstractString,Regex},
+    project_tree::AbstractDict)
+```
+
+An example of the use of the `depending_projects()` function may look as follows:
+
+```julia-repl
+julia> using IntegrationTests
+[ Info: Precompiling IntegrationTests [6be7cfa2-1838-408e-bc49-3a824ac3a1fb]
+
+julia> using Pkg
+
+julia> Pkg.activate(".ci/example_project/MetaTestPkg/")
+  Activating project at `~/projects/IntegrationTests.jl/.ci/example_project/MetaTestPkg`
+
+julia> depending_projects("MyPkgFields", r"MyPkg*")
+2-element Vector{String}:
+ "MyPkgProcesses"
+ "MyPkgMain"
+
+julia>
+```
+
+# Package Filter
+
+The package filter restricts the search space of the dependency graph. All packages and dependencies of packages must match the filter, otherwise they are ignored for the result and the search. If we set the package filter to `r"^MyPkg*"`, the packages `MyPkgMain`, `MyPkgA`, `MyPkgB` and `MyPkgC` are traversed in the following diagram. `ExternalDep1` and `ExternalDep2` are ignored.
+
+```mermaid
+graph TD
+    MyPkgMain -->|using| MyPkgA
+    MyPkgMain -->|using| MyPkgB
+    MyPkgMain -->|using| ExternDep1
+    MyPkgA -->|using| MyPkgC
+    MyPkgA -->|using| ExternDep2
+```
+
+The idea behind the package filter is that there are dependencies in the dependency graph that we cannot change directly. Let's assume we are the maintainer of all packages starting with `MyPkg` and the following dependency graph is given:
+
+```mermaid
+graph TD
+    MyPkgMain -->|using| MyPkgA
+    MyPkgMain -->|using| MyPkgB
+    MyPkgMain -->|using| ExternDep1 -->|using| MyPkgA
+    MyPkgA -->|using| MyPkgC
+    MyPkgA -->|using| ExternDep2
+```
+
+With the package filter `r"*"`, which means traverse all dependencies, the result would be `["MyPkgMain", "ExternDep1"]` if we are searching for `MyPkgA`. We are not the maintainer of `ExternDep1`, therefore we can only hardly introduce changes in `ExternDep1` and it make less sense to test `ExternDep1` with our code changes.
+
+With the package filter `r "*"`, which means that all dependencies are traversed, the result would be `["MyPkgMain", "ExternDep1"]` if we search for `MyPkgA`. We are not the maintainer of `ExternDep1`, so we can hardly change code in `ExternDep1` and it makes little sense to test `ExternDep1` with our code changes.
```

Comment:
same paragraph again?

Reply:
Ops, now you know my dark secret :sweat_smile: I use deepl to improve my sentences. Therefore I copy the sentence to deepl and the modified version back. That's a left over, which I need to remove.